src/neo/SmartContract/Helper.cs

```diff
@@ -115,6 +116,77 @@ public static bool IsSignatureContract(this byte[] script)
             return true;
         }
 
+        private static readonly byte[] balanceOf = Encoding.UTF8.GetBytes("balanceOf");
+
+        public static bool IsSolidTransfer(this byte[] script)
+        {
+            if (script.Length < 10) return false;
+            int index;
+
+            if (script[0] >= (byte)OpCode.PUSHM1 && script[0] <= (byte)OpCode.PUSH16)
+            {
+                if (script.Length != 65) return false;
+                index = 1;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT8)
+            {
+                if (script.Length != 66) return false;
+                index = 2;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT16)
+            {
+                if (script.Length != 67) return false;
+                index = 3;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT32)
+            {
+                if (script.Length != 69) return false;
+                index = 5;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT64)
+            {
+                if (script.Length != 73) return false;
+                index = 9;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT128)
+            {
+                if (script.Length != 81) return false;
+                index = 17;
+            }
+            else if (script[0] == (byte)OpCode.PUSHINT256)
+            {
+                if (script.Length != 97) return false;
+                index = 33;
+            }
+            else
+            {
+                // Other pushes
```

Comment:
Maybe we can try to simplify here, by restricting to some general PushInt operation... do we have that?

Reply:
The problem is that script builder generate the most optimized one